venom/rpc/reflect/openapi.py

```diff
@@ -1,6 +1,195 @@
+TYPES = {
+    'str': 'string',
+    'int': 'integer',
+    'float': 'double',
+    'bool': 'boolean',
+}
+
+PATH_PARAMETER = {
+    'is_in': 'path',
+    'required': True
+}
+
+BODY_PARAMETER = {
+    'is_in': 'body',
+}
+
+QUERY_PARAMETER = {
+    'is_in': 'query',
+}
+
+
+def description_param(field: Field) -> dict:
+    if DESCRIPTION in field.options:
+        return {DESCRIPTION: field.options.get(DESCRIPTION)}
+    return {}
+
+
+@singledispatch
+def type_message(field: Field) -> TypeMessage:
+    f_type = field.type.__name__
+    if f_type not in TYPES:
+        return TypeMessage(ref=reference_string(field.type))
+    return TypeMessage(
+        type=TYPES[f_type],
+        **description_param(field)
+    )
+
+
+@type_message.register(RepeatField)
+def type_message_repeat(field: RepeatField) -> TypeMessage:
+    return TypeMessage(
+        type=ARRAY,
+        items=type_message(field.items),
+        **description_param(field)
+    )
+
+
+@type_message.register(MapField)
+def type_message_map(field: MapField) -> TypeMessage:
+    return TypeMessage(
+        type='object',
+        additionalProperties=type_message(field.values),
+        **description_param(field)
+    )
+
+
+def reference_string(message: Message) -> str:
+    name = message.__meta__.name
+    if name == 'Empty':
+        return ''
+    return '#/{}/{}'.format(DEFINITIONS, name)
+
+
+def parameter_common(field: Field) -> dict:
+    result = {'name': field.name}
+    f_type = type_message(field)
+    protocol = JSON(type(f_type))
+    result.update(protocol.encode(f_type))
+    return result
+
+
+def parameters_path(method: Method) -> list:
+    fields = [getattr(method.request, f) for f in method.http_path_parameters()]
+    return [{**PATH_PARAMETER, **parameter_common(f)} for f in fields]
+
+
+def parameters_query(method: Method) -> list:
+    fields = [getattr(method.request, f)
+              for f in method.http_field_locations()[HTTPFieldLocation.QUERY]]
+    return [{**QUERY_PARAMETER, **parameter_common(f)} for f in fields]
+
+
+def parameters_body(method: Method) -> list:
+    body_fields = method.http_field_locations()[HTTPFieldLocation.BODY]
+    if not body_fields:
+        return []
+    fields = {f: getattr(method.request, f) for f in body_fields}
+    if fields == method.request.__fields__:
+        param = dict(
+            name=method.request.__meta__.name,
+            schema=TypeMessage(ref=reference_string(method.response))
+        )
+    else:
+        param = dict(
+            name=method.name + '_body',
+            schema=fields_message(fields.values())
+        )
+    return [{**BODY_PARAMETER, **param}]
+
+
+def to_message(param_data: dict) -> ParameterMessage:
+    return ParameterMessage(
+        **{k: param_data[k] for k in field_names(ParameterMessage) if k in param_data}
+    )
+
+
+def parameters_messages(m: Method) -> list:
+    return list(map(to_message, parameters_body(m) + parameters_path(m) + parameters_query(m)))
```

Comment:
This is very hard to understand. Avoid `map()` and use comprehensions where possible.

I am not sure this needs an extra function since it is only used in one place and is a one-liner.